### PR TITLE
fix: jdbc4 plugin remoteServiceName

### PR DESCRIPTION
**Pull request by lanxenet**
Correct the issue of span.remoteServiceName in the JDBC plugin always being for MySQL. Modify it to be displayed based on the connection string, such as mysql:database_name, postgresql:database_name, oracle:database_name.

### CODE REVIEW

File: config/src/main/java/com/megaease/easeagent/config/ConfigFactory.java [hunk 1]
```diff
@@ -95,6 +95,7 @@ public static GlobalConfigs loadConfigs(String pathname, ClassLoader loader) {
         // override by user special config file
         if (StringUtils.isNotEmpty(pathname)) {
             GlobalConfigs configsFromOuterFile = ConfigLoader.loadFromFile(new File(pathname));
+            LOGGER.info("Loaded user special config file: {}", pathname);
```
Comment (observeralone) [inline config/src/main/java/com/megaease/easeagent/config/ConfigFactory.java:98]: If not necessary, please refrain from adding info logs to the framework.  If deemed necessary, please use debug logs.

File: plugins/jdbc/src/main/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptor.java [hunk 2]
```diff
@@ -84,19 +83,20 @@ public void doBefore(MethodInfo methodInfo, Context context) {
...
     }
 
-    public String remoteServiceName(String database) {
-        if (StringUtils.isEmpty(database)) {
-            return "mysql";
-        } else {
-            return "mysql-" + database;
-        }
+    public String remoteServiceName(DatabaseInfo info) {
+        return String.format("%s:%s", info.getDatabaseType(), info.getDatabase());
```
Comment (observeralone) [inline plugins/jdbc/src/main/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptor.java:94]:
```suggestion
        return String.format("%s-%s", info.getDatabaseType(), info.getDatabase());
```
If not necessary, please maintain the formatting.

File: plugins/jdbc/src/main/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptor.java [hunk 3]
```diff
@@ -84,19 +83,20 @@ public void doBefore(MethodInfo methodInfo, Context context) {
...
+//        if (StringUtils.isEmpty(info.getServiceName())) {
+//            return "mysql";
+//        } else {
+//            return "mysql-" + database;
+//        }
```
Comment (observeralone) [inline plugins/jdbc/src/main/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptor.java:99]:
```suggestion
```
If you are certain that this piece of code is not needed, please remove it to keep things tidy.

**Review by observeralone — APPROVED** [review]

**Review by observeralone — APPROVED** [review]

File: plugins/jdbc/src/test/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptorTest.java [hunk 4]
```diff
@@ -82,7 +82,7 @@ public void doBefore() throws SQLException {
         assertEquals("database", reportSpan.tag(JdbcStmTracingInterceptor.SPAN_LOCAL_COMPONENT_TAG_NAME));
         assertEquals(TestUtils.URI, reportSpan.tag(JdbcStmTracingInterceptor.SPAN_URL));
         assertEquals(Type.DATABASE.getRemoteType(), reportSpan.tag(MiddlewareConstants.TYPE_TAG_NAME));
-        assertEquals("mysql-" + TestUtils.DATABASE, reportSpan.remoteServiceName());
+        assertEquals(TestUtils.DATABASE_TYPE + ":" + TestUtils.DATABASE, reportSpan.remoteServiceName());
```
Comment (observeralone) [inline plugins/jdbc/src/test/java/com/megaease/easeagent/plugin/jdbc/interceptor/tracing/JdbcStmTracingInterceptorTest.java:85]:
```suggestion
        assertEquals(TestUtils.DATABASE_TYPE + "-" + TestUtils.DATABASE, reportSpan.remoteServiceName());
```

**Review by observeralone — APPROVED** [review]

**Review by landyking — APPROVED** [review]

**Pull request merged**